packages/common/src/paginator.js

```diff
@@ -231,4 +247,25 @@ streamRouter.runAsStream_ = function(parsedArguments, originalMethod) {
   return limiter.stream;
 };
 
-module.exports = streamRouter;
+/**
+ * Wraps original method in auto paginated function and provides a getter
+ * for the original in the event that we wish to streamify that same method
+ * later.
+ *
+ * @param {function} originalMethod - The original method.
+ * @return {function} wrapped - The wrapped method.
+ */
+paginator.wrap_ = function(originalMethod) {
```

Comment:
Currently, we overwrite a given method with the "paginator" version. At that point in the code, we still have the original, so how about we just put the originals in a different place, i.e.

```js
Class.prototype[originalMethodName + '_'] = originalMethod;
```

then call that from streamify(). Will that work? If so, we can probably leave the "wrap" behavior where it was.